src/app/(admin)/new_dashboard/guides/hooks/sortGuides.ts

```diff
@@ -0,0 +1,14 @@
+import { Guide } from "@/api/types/Guide";
+
+export const sortByCriteria = (guides: Guide[], sortCriteria: string): Guide[] => {
+    return guides.sort((a, b) => {
+        if (sortCriteria === "rating") {
+            const avgRatingA = a.reviews?.reduce((acc, review) => acc + review.rating, 0) ?? 1;
```

Comment:
again, this is the same function twice, make a util function to calculate the avg rating

Reply:
ok